core/ledger/kvledger/kv_ledger.go

```diff
@@ -78,24 +69,17 @@ func newKVLedger(
 		blockAPIsRWLock: &sync.RWMutex{},
 	}
 
+	ccInfoProvider := lgrProvider.initializer.DeployedChaincodeInfoProvider
 	btlPolicy := pvtdatapolicy.ConstructBTLPolicy(&collectionInfoRetriever{ledgerID, l, ccInfoProvider})
 
-	if err := l.initTxMgr(
-		versionedDB,
-		stateListeners,
-		btlPolicy,
-		bookkeeperProvider,
-		ccInfoProvider,
-		customTxProcessors,
-		hasher,
-	); err != nil {
+	err := l.initTxMgr(versionedDB, lgrProvider.stateListeners, btlPolicy, lgrProvider)
```

Comment:
you're already passing in the `lgrProvider` you can pull the `lgrProvider.stateListeners` off the object in the callee instead of passing it down explicitly.

Reply:
I am not against clubbing the parameters to reduce the number of parameters. My objection is to the way it is achieved here. i.e., pass down the `provider` instance much deeper in the code where we are supposed to have the code specific to one ledger instance. This is overusing the `provider`, which is not merely a holder of objects but has an associated behavior as well.